src/components/test/common-tests.js

```diff
@@ -237,3 +238,40 @@ export function testSimpleComponent(Component) {
     });
   });
 }
+
+/**
+ * Common tests for simple design components
+ *
+ * @param {TransitionComponent} Component
+ * @param {CommonTestOpts} opts
+ */
+export function testTransitionComponent(
+  Component,
+  { componentName, createContent = createComponent } = {}
+) {
+  const displayName = componentName ?? Component.displayName ?? Component.name;
+
+  describe(`Common transition component functionality for ${displayName}`, () => {
+    it('renders', () => {
+      const wrapper = createComponent(Component);
+      assert.isTrue(wrapper.find(Component).exists());
+    });
+
+    ['in', 'out'].forEach(direction => {
+      it('should handle unmounting on either direction', () => {
+        const wrapper = createComponent(Component, { direction });
+        wrapper.setProps({ direction: direction === 'in' ? 'out' : 'in' });
+        wrapper.unmount();
+      });
+    });
```

Comment:
Do we need a way to test that the `TransitionComponent` will invoke the `onTransitionEnd` callback? I know that this introduces a time component into the testing...

Reply:
Good point. Actually, I think the purpose of this test was probably that, and it got lost at some point (I migrated it from `client` first, and then here. I may have messed it up 😅)

Reply:
I have changed the test. No it covers the invocation to `onTransitionEnd`.